src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestReferenceCounts2.ll

```diff
@@ -49,20 +49,5 @@ else__1:                                          ; preds = %entry
   ret { %String*, %Array* }* %s
 
 continue__1:                                      ; No predecessors!
```

Comment:
I know it's not part of this change per-se, but why are we generating a block with no predecessors? This would get trimmed by CFG simplification anyway, so it's not a big deal, but it strikes me as odd. Perhaps an investigation for later?

Reply:
This is an artifact of how if statements are handled. Basically we add the block before we know it is needed - now I can't remember exactly the details, but if I remember correctly this was contributed by someone else as part of a bug fix, and it indeed seems the simplest and most robust solution.